operators/pkg/instance-controller/containers_test.go

```diff
@@ -0,0 +1,397 @@
+					Spec:       corev1.PersistentVolumeClaimSpec{},
+					Status:     corev1.PersistentVolumeClaimStatus{},
+				}
+				existing.SetCreationTimestamp(metav1.NewTime(time.Now()))
+				clientBuilder.WithObjects(&existing)
+			})
+
+			It("The PVC should be present and have the common attributes", func() {
+				Expect(reconciler.Get(ctx, objectName, &pvc)).To(Succeed())
+				Expect(pvc.GetLabels()).To(Equal(forge.InstanceObjectLabels(nil, &instance)))
+				Expect(pvc.GetOwnerReferences()).To(ContainElement(ownerRef))
+			})
+
+			It("The PVC should be present and have unmodified specs", func() {
+				Expect(reconciler.Get(ctx, objectName, &pvc)).To(Succeed())
+				Expect(pvc.Spec).To(Equal(corev1.PersistentVolumeClaimSpec{}))
+			})
+		})
+
+		When("the deployment is already present", func() {
+			BeforeEach(func() {
+				existing := appsv1.Deployment{
+					ObjectMeta: forge.NamespacedNameToObjectMeta(objectName),
+					Spec:       appsv1.DeploymentSpec{},
+					Status:     appsv1.DeploymentStatus{ReadyReplicas: 0},
+				}
+				existing.SetCreationTimestamp(metav1.NewTime(time.Now()))
+				clientBuilder.WithObjects(&existing)
+			})
+
+			It("Should not return an error", func() { Expect(err).ToNot(HaveOccurred()) })
+
+			It("The deployment should still be present and have the common attributes", func() {
+				Expect(reconciler.Get(ctx, objectName, &deploy)).To(Succeed())
+				Expect(deploy.GetLabels()).To(Equal(forge.InstanceObjectLabels(nil, &instance)))
+				Expect(deploy.GetOwnerReferences()).To(ContainElement(ownerRef))
+			})
+
+			It("The deployment should still be present and have unmodified specs", func() {
+				Expect(reconciler.Get(ctx, objectName, &deploy)).To(Succeed())
+				// Here we overwrite the replicas value, as it is checked in a different It clause.
+				deploy.Spec.Replicas = nil
+				Expect(deploy.Spec).To(Equal(appsv1.DeploymentSpec{}))
+			})
+
+			It("Should set the correct instance phase", func() {
+				Expect(instance.Status.Phase).To(BeIdenticalTo(clv1alpha2.EnvironmentPhaseStarting))
+			})
+
+			Context("The instance is running", func() {
+				BeforeEach(func() { instance.Spec.Running = true })
+
+				It("The deployment should be present and with one replica", func() {
+					Expect(reconciler.Get(ctx, objectName, &deploy)).To(Succeed())
+					Expect(deploy.Spec.Replicas).To(PointTo(BeNumerically("==", 1)))
+				})
+			})
+
+			Context("The instance is not running", func() {
+				BeforeEach(func() { instance.Spec.Running = false })
+
+				It("The deployment should be present and with no replicas", func() {
+					Expect(reconciler.Get(ctx, objectName, &deploy)).To(Succeed())
+					Expect(deploy.Spec.Replicas).To(PointTo(BeNumerically("==", 0)))
+				})
+			})
+		})
+	})
+})
+
+// deploymentSpecCleanup removes resources which may conflict because of reordering
+// or internal representation differences.
+func deploymentSpecCleanup(ds *appsv1.DeploymentSpec) {
```

Comment:
To avoid polluting the `_test` package, I would define this as a variable within the outermost describe block.